src/olympia/addons/forms.py

```diff
@@ -53,12 +53,27 @@ def clean_addon_name(name, instance=None, addon_type=None):
     if not addon_type:
         addon_type = instance.type
 
-    id = reverse_name_lookup(name, addon_type)
+    matches = reverse_name_lookup(name, addon_type)
+
+    if matches is not None:
+        addon_id = matches.keys()[0]
+        locales = matches.values()[0]
+
+        if addon_id and (not instance or addon_id != instance.id):
+            # If we get an id and either there's no instance
+            # or the instance.id != id.
+            if id and (not instance or id != instance.id):
+                if isinstance(name, dict):
+                    # Return locale specific message
+                    msg = _(
+                        'This name is already in use for {}. '
+                        'Please choose another.'.format(
+                            ', '.join(locales)))
```

Comment:
Shouldn't we transform the locales to their full names ? i.e. instead of "This name is already in use for en-us, fr" say "This name is already in use for English, French"

Also, you're missing a `u''` prefix on that string (which will be even more important if you implement my suggestion, since you can end up with a parameter containing non-ascii chars. a test would be nice)